packages/algorand-js/src/runtime/account.ts

```diff
@@ -75,25 +75,23 @@ export class StoreAccountImpl implements StoreAccount {
    * @param key: key to fetch value of from local state
    * @param value: key to fetch value of from local state
    */
-  updateLocalState (appId: number, key: Uint8Array, value: StackElem): AppLocalState[] {
-    const localState = this.appsLocalState;
+  updateLocalState (appId: number, key: Uint8Array, value: StackElem): AppLocalState {
+    const localState = this.appsLocalState.get(appId);
     const data = getKeyValPair(key, value); // key value pair to put
 
-    for (const l of localState) {
-      if (l.id === appId) { // find appId
-        const localApp = l["key-value"];
-        const idx = localApp.findIndex(schema => compareArray(schema.key, key));
-
-        if (idx === -1) {
-          localApp.push(data); // push new pair if key not found
-        } else {
-          localApp[idx].value = data.value; // update value if key found
-        }
-        l["key-value"] = localApp; // save updated state
+    const localApp = localState?.["key-value"];
+    if (localState && localApp) {
+      const idx = localApp.findIndex(schema => compareArray(schema.key, key));
 
-        assertValidSchema(l["key-value"], l.schema); // verify if updated schema is valid by config
-        return localState;
+      if (idx === -1) {
+        localApp.push(data); // push new pair if key not found
+      } else {
+        localApp[idx].value = data.value; // update value if key found
       }
+      localState["key-value"] = localApp; // save updated state
+
+      assertValidSchema(localState["key-value"], localState.schema); // verify if updated schema is valid by config
```

Comment:
`"key-value"` is used few times. Please use `const` value for that, instead of strings.

Reply:
Okay